exercises/047_methods.zig

```diff
@@ -2,41 +2,44 @@
 // Help! Evil alien creatures have hidden eggs all over the Earth
 // and they're starting to hatch!
 //
-// Before you jump into battle, you'll need to know four things:
+// Before you jump into battle, you'll need to know three things:
 //
-// 1. You can attach functions to structs:
+// 1. You can attach functions to structs (and other "type definitions"):
 //
 //     const Foo = struct{
 //         pub fn hello() void {
 //             std.debug.print("Foo says hello!\n", .{});
 //         }
 //     };
 //
-// 2. A function that is a member of a struct is a "method" and is
-//    called with the "dot syntax" like so:
+// 2. A function that is a member of a struct is "namespaced" within
+//    that struct and is called by specifying the "namespace" and then
+//    using the "dot syntax":
 //
 //     Foo.hello();
 //
-// 3. The NEAT feature of methods is the special parameter named
-//    "self" that takes an instance of that type of struct:
+// 3. The NEAT feature of these functions is that if they take either
+//    an instance of the struct or a pointer to an instance of the struct
+//    then they have some syntax sugar:
 //
 //     const Bar = struct{
-//         number: u32,
-//
-//         pub fn printMe(self: Bar) void {
-//             std.debug.print("{}\n", .{self.number});
-//         }
+//         pub fn a(self: Bar) void { _ = self; }
+//         pub fn b(this: *Bar, other: u8) void { _ = this; _ = other; }
+//         pub fn c(bar: *const Bar) void { _ = bar; }
 //     };
 //
-//    (Actually, you can name the first parameter anything, but
-//    please follow convention and use "self".)
+//    var bar = Bar{};
+//    bar.a() // is equivalent to Bar.a(bar)
+//    bar.b(3) // is equivalent to Bar.b(&bar, 3)
+//    bar.c() // is equivalent to Bar.c(&bar)
 //
-// 4. Now when you call the method on an INSTANCE of that struct
-//    with the "dot syntax", the instance will be automatically
-//    passed as the "self" parameter:
+//    Notice that the name of the parameter doesn't matter. Some use
+//    self, others use a lowercase version of the type name, but feel
+//    free to use whatever is most appropriate.
 //
-//     var my_bar = Bar{ .number = 2000 };
-//     my_bar.printMe(); // prints "2000"
+//    Effectively, the method syntax sugar just does this transformation:
+//        thing.function(args);
+//        @TypeOf(thing).function(thing, args);
```

Comment:
This explanation is definitely too advanced at this stage. We don't cover the built-in functions until 065. But I get what you're saying. Can we put this into words?